### DISCUSSION TITLE
Request limiting and anti-fingerprinting

### DISCUSSION THREAD

**Discussion by ghost** [Eloston/ungoogled-chromium#1844] [General]
Sometimes when I search for something on github, I get an error message saying my request rate limit was exceeded, even if it's the first time I opened github that day.
For context, I often browse github in private mode, and I have enabled several of the privacy enhancing features I found (anti-fingerprinting included).
I don't really have a problem with this, and I wouldn't disable anti-fingerprinting in any case. I'm just wondering if anyone has observed similar issues and can confirm that they are, indeed, due to this.

Edit:
After further searching, I came across this [github doc page](https://docs.github.com/en/rest/reference/search#rate-limit) about their REST api.
Basically, when using the api, unauthenticated users are limited to 10 search requests per minute. I imagine a similar limitation applies to the web interface.
The most reliable way to enforce such a limitation, as far as I understand it, would be through fingerprinting (using the IP address among other things). And so, anti-fingerprinting, when effective, would conceivably confuse such a mechanism.

Edit 2:
After yet more searching, I discovered several projects on github itself that use some form of fingerprinting for this very purpose, mostly based on the the IP address, like [Laravel](https://github.com/laravel/framework/pull/19807).
Of course, given the availability of monstrosities like filthystalker.js, web apps can easily fingerprint their users with much more precision, and I would imagine many of them do.

**Comment by networkException** [comment]
I use GitHub extensively with all my usual ad blocking and tracking preventions enabled and have never run into a rate limit. Maybe check that you don't have a token that's being used in some automated manner?

**Reply by ghost** [reply]
No, I don't have any such automated mechanism set up.
However, with my ISP, our router's public address changes on almost every reboot. Maybe github handles such IP ranges differently. In addition, this only ever happened when I was logged off.
Also, could you tell me if you have whitelisted cookies for github, and if you ever perform site-wide searches in private mode ?

**Reply by networkException** [reply]
I only allow first party cookies, I can't really say anything about private mode as I don't have a use case for that unless I need to use a different GitHub account (which is rare)

**Reply by ghost** [reply]
I see. Thanks for info in any case.
